x/fibre/blob.go

```diff
@@ -0,0 +1,307 @@
+package fibre
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"runtime"
+
+	"github.com/celestiaorg/rsema1d"
+	"github.com/celestiaorg/rsema1d/field"
+)
+
+// ErrBlobTooLarge is returned when the blob size exceeds MaxBlobSize.
+var ErrBlobTooLarge = errors.New("blob size exceeds maximum allowed size")
+
+// Commitment is a commitment to a blob.
+// TODO(@Wondertan): merge with rsema1d.Commitment once it has these methods.
+type Commitment rsema1d.Commitment
+
+// UnmarshalBinary decodes a [Commitment] from bytes.
+func (c *Commitment) UnmarshalBinary(data []byte) error {
+	if len(data) != 32 {
+		return fmt.Errorf("commitment must be 32 bytes, got %d", len(data))
+	}
+	copy(c[:], data)
+	return nil
+}
+
+// String returns the hex-encoded string representation of the commitment.
+func (c Commitment) String() string {
+	return hex.EncodeToString(c[:])
+}
+
+// BlobConfig contains configuration for erasure coding and data handling.
+type BlobConfig struct {
+	// OriginalRows is the number of original rows before erasure coding.
+	// Here we use explicit naming: OriginalRows (K in rsema1d) and ParityRows (N in rsema1d).
+	// SPECDO: The spec uses N to represent total rows (original + parity), while rsema1d defines N as parity only.
+	OriginalRows int
+	// ParityRows is the number of parity rows added by erasure coding.
+	// Total rows = OriginalRows + ParityRows.
+	ParityRows int
+	// RowSizeMin is the minimum row size in bytes.
+	RowSizeMin int
+	// MaxBlobSize is the maximum allowed blob size.
+	MaxBlobSize int
+	// BlobVersion is the version of the row format.
+	BlobVersion uint8
```

Comment:
[uber nit] to clarify these are for Fibre blobs. I guess it's somewhat implied by this living in the x/fibre package but we already have constants in celestia-app for the existing blobs so a Fibre in the name would help differentiate.


```suggestion
	// MaxFibreBlobSize is the maximum allowed Fibre blob size.
	MaxFibreBlobSize int
	// FibreBlobVersion is the version of the row format.
	FibreBlobVersion uint8
```

Reply:
>  I guess it's somewhat implied by this living in the x/fibre package but we already have constants in celestia-app for the existing blobs so a Fibre in the name would help differentiate.

If we do it for blob then we should also do it for all the mentions of Row, like RowSize or MaxRowSize, etc, because row do also exist in the core protocol. I believe the `fibre` as pkg is a sufficient domain separator and adding it again within the code for everything that might be similar with core protocol is superfluous.

Reply:
I don't think RowSize or MaxRowSize need a Fibre prefix because celestia-app doesn't have constants for those already so there isn't a real risk of mixing those up with existing terms in celestia-app.

On the other hand, there are existing blobs in celestia-app and in celestia-app v3.x there was even an ante handler [here](https://github.com/celestiaorg/celestia-app/blob/4dd92890ac0a50d163627053d6e0323c9b5ba6a9/x/blob/ante/max_total_blob_size_ante.go#L46-L57) which is really similar to the `MaxBlobSize` idea expressed here.

Reply:
> I don't think RowSize or MaxRowSize need a Fibre prefix because celestia-app doesn't have constants for those already so there isn't a real risk of mixing those up with existing terms in celestia-app.

So why wouldn't we add the prefix to the constants we put them together with constants for core protocol? This is just a config that is agnostic to what protocol is and doesn't know if there is some other core params with similar names. For this config, prefixing Fibre is redundant. For the place which instaniates the config - it will be not, so we can keep it there.

Reply:
Resolving b/c I'm not blocking on adding the Fibre in the error message or name, it was just a suggestion.